### PR TITLE
feat: self supervised learning integration

### PR DESCRIPTION
**Pull request by bwanglzu**
this PR connects all components for self-supervised learning v1.

### CODE REVIEW

File: finetuner/tuner/augmentation.py [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
 from docarray import Document
 
 
 def vision_preprocessor(
     doc: Document,
     height: int = 224,
     width: int = 224,
-    channel_axis: int = -1,
+    default_channel_axis: int = -1,
+    target_channel_axis: int = 0,
```
Comment (gmastrapas) [inline finetuner/tuner/augmentation.py:10]: update docstring as well

File: finetuner/tuner/pytorch/datasets.py [hunk 2]
```diff
@@ -29,3 +29,10 @@ def __getitem__(self, ind: int) -> Tuple['DocumentContentType', Tuple[int, int]]
         content, label = super().__getitem__(ind)
         content = _make_blob_writable(content)
         return (content, label)
+
+
+class PytorchInstanceDataset(InstanceDataset, PytorchDataset):
+    def __getitem__(self, ind: int) -> Tuple['DocumentContentType', Tuple[int, int]]:
```
Comment (gmastrapas) [inline finetuner/tuner/pytorch/datasets.py:35]: `Tuple['DocumentContentType', int]` ?

**Review by gmastrapas — CHANGES_REQUESTED** [review]
Some comments

**Comment by gmastrapas** [comment]
So basically if I pass unlabeled data (no tags nor matches) the ssl kicks in? What about custom `augment_fn`?

**Comment by tadejsv** [comment]
@gmastrapas There is no `augment_fn`, you can use the `preprocess_fn` for that

**Comment by bwanglzu** [comment]
i need to refactor `vision_preprocessor` a bit, hold on :)

**Review by tadejsv — APPROVED** [review]
Looks good

**Review by gmastrapas — APPROVED** [review]
LGTM!

**Pull request merged**